### PR TITLE
RUN-4252/improve event handling

### PR DESCRIPTION
**Pull request by pbaize**

### CODE REVIEW

File: src/api/frame/frame.ts [hunk 1]
```diff
@@ -38,12 +38,11 @@ export default class _FrameModule extends Base {
  * @class
  * @alias Frame
  */
-// @ts-ignore: return types incompatible with EventEmitter (this)
 // tslint:disable-next-line
 export class _Frame extends EmitterBase {
 
     constructor(wire: Transport, public identity: Identity) {
-        super(wire);
+        super(wire, ['frame', identity.uuid]);
```
Comment (rdepena) [inline src/api/frame/frame.ts]: frame will require uuid/name

**Comment by rdepena** [comment]
Can we get test-runner results with a replaced core for these changes as well ?

File: src/api/events/eventAggregator.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const buildLocalPayload = (rawPayload: any): NotificationCallback => {
+    const { payload: { message }, type } = rawPayload;
+
+    const payload: NotificationCallback = {};
+
+    switch (type) {
+        case 'message': payload.message = message;
+            break;
+        case 'show':
+        case 'error':
```
[the remark this answers is not in the document]
Comment (pbaize) [reply src/api/events/eventAggregator.ts]: I took this from the previous implementation

**Comment by pbaize** [comment]
Test results in core:
[Windows 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b7b3aea6107b45e6d1ebade) (getInfo timed out)
[Windows 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b7b394a6107b45e6d1ebadd)

**Pull request merged**